src/Aspire.Hosting/api/Aspire.Hosting.cs

```diff
@@ -621,6 +717,48 @@ public ConnectionStringReference(IResourceWithConnectionString resource, bool op
         System.Threading.Tasks.ValueTask<string?> IValueProvider.GetValueAsync(System.Threading.CancellationToken cancellationToken) { throw null; }
     }
 
+    public sealed partial class ContainerDirectory : ContainerFileSystemItem
+    {
+        public System.Collections.Generic.IEnumerable<ContainerFileSystemItem> Entries { get { throw null; } set { } }
+    }
+
+    public sealed partial class ContainerFile : ContainerFileSystemItem
+    {
+        public string? Contents { get { throw null; } set { } }
+    }
+
+    [System.Diagnostics.DebuggerDisplay("Type = {GetType().Name,nw}, DestinationPath = {DestinationPath}")]
+    public sealed partial class ContainerFileSystemCallbackAnnotation : IResourceAnnotation
+    {
+        public required System.Func<ContainerFileSystemCallbackContext, System.Threading.CancellationToken, System.Threading.Tasks.Task<System.Collections.Generic.IEnumerable<ContainerFileSystemItem>>> Callback { get { throw null; } init { } }
+
+        public int DefaultGroup { get { throw null; } init { } }
+
+        public int DefaultOwner { get { throw null; } init { } }
+
+        public required string DestinationPath { get { throw null; } init { } }
+
+        public System.IO.UnixFileMode? Umask { get { throw null; } set { } }
```

Comment:
@danegsta - does this Umask apply in addition to, or in replace of, the umask in the docker container?

For example, if the docker container itself had a higher umask than 022 (say 027) and this Umask was set to 022, what is the final umask applied to the file system items?

Reply:
The container umask isn’t applied, only this umask is used for the added files. We’re writing the final permission mask into a tar file which gets streamed over stdin via the `cp` command while the container is created but not running, so the permissions to be applied are determined before the file is sent to the container.